driver-core/src/main/java/com/datastax/driver/core/NettyUtil.java

```diff
@@ -0,0 +1,86 @@
+/*
+ *      Copyright (C) 2012-2014 DataStax Inc.
+ *
+ *   Licensed under the Apache License, Version 2.0 (the "License");
+ *   you may not use this file except in compliance with the License.
+ *   You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *   Unless required by applicable law or agreed to in writing, software
+ *   distributed under the License is distributed on an "AS IS" BASIS,
+ *   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *   See the License for the specific language governing permissions and
+ *   limitations under the License.
+ */
+package com.datastax.driver.core;
+
+import java.util.Locale;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * A set of utilities related to the underlying Netty layer.
+ */
+class NettyUtil {
+
+    private static final boolean FORCE_NIO = SystemProperties.getBoolean("com.datastax.driver.FORCE_NIO", false);
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(NettyUtil.class);
+
+    private static final boolean SHADED;
+
+    private static final boolean USE_EPOLL;
+
+    private static final String NON_SHADED_EPOLL = "io.netty.channel.epoll.EpollEventLoopGroup";
+
+    private static final String SHADED_EPOLL = "com.datastax.shaded.netty.channel.epoll.EpollEventLoopGroup";
+
+    static {
+        boolean shaded;
+        try {
+            Class.forName("io.netty.channel.Channel");
+            shaded = false;
+        } catch (ClassNotFoundException e) {
+            try {
+                Class.forName("com.datastax.shaded.netty.channel.Channel");
+                shaded = true;
+            } catch (ClassNotFoundException e1) {
+                throw new AssertionError("Cannot locate Netty classes in the classpath", e1);
+            }
+        }
+        SHADED = shaded;
+        boolean useEpoll;
+        try {
+            if(SHADED)
```

Comment:
I'm adding this if block because it seems to me that the shaded version of the driver would not detect properly the presence of EpollEventLoopGroup. Am I mistaken here?
Generally speaking, does the shade plugin take care of shading things like Class.forName("io.netty.*")? I've seen a lot of these in Netty code.